### PR TITLE
Behavior tree special attack integration

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
Continuing #39810 
tl;dr the monster AI code is inflexible and difficult to extend, this is part of the process of migrating the whole thing to use behavior trees to make it more legible and flexible.

#### Describe the solution
This addresses most of the major blockers to onboarding special attacks to behavior trees.
Extends the behavior predicate API to accept an optional argument and use it to specify a special id to check for. This avoids having to add a predicate for each special, and I anticipate this sort of thing coming up repeatedly.
Adds predicates as needed to determine whether the target special is triggerable.
Added code to inject the behavior tree definition into monster types that have the special.
Added code to invoke the special attack if selected.

#### Testing
Extended the unit test I added previously.
Need to spawn in a locust and induce it to nibble some plants.

#### Additional context
Great resource for behavior trees: http://www.gameaipro.com/GameAIPro/GameAIPro_Chapter06_The_Behavior_Tree_Starter_Kit.pdf

### CODE REVIEW

**Comment by kevingranade** [comment]
Tested out fine, the only thing I might add here is documentation on how behavior -> special attack correspondence works, but if you merge it before that goes in I'll just add it in another PR.

**Comment by kevingranade** [comment]
Hooray, weird crash in tests I didn't encounter locally.

**Comment by kevingranade** [comment]
The crash is legit. I've resolved it now it was a data error and I've also corrected the issue that escalated the data issue to be a crash.

File: tests/behavior_test.cpp [hunk 1]
```diff
@@ -183,18 +185,28 @@ TEST_CASE( "check_npc_behavior_tree", "[npc][behavior][!mayfail]" )
...
-    monster_goals.add( &string_id<behavior::node_t>( "monster_special" ).obj() );
-    monster &test_monster = spawn_test_monster( "mon_zombie", { 5, 5, 0 } );
+    monster_goals.add( test_monster.type->get_goals() );
+
     for( const std::string &special_name : test_monster.type->special_attacks_names ) {
         test_monster.reset_special( special_name );
     }
-    behavior::monster_oracle_t oracle( &test_monster );
     CHECK( monster_goals.tick( &oracle ) == "idle" );
+    for( const tripoint &near : g->m.points_in_radius( monster_location, 1 ) ) {
```
Comment (ZhilkinSerg) [inline tests/behavior_test.cpp:200]: It is ridiculous, but [you can't use `near` as variable name in VS](https://stackoverflow.com/questions/8948493/visual-studio-doesnt-allow-me-to-use-certain-variable-names).

image

Comment (kevingranade) [reply tests/behavior_test.cpp:200]: Gah, yes I've seen that before, I need to add a lgtm filter or something to warn on this and other VS reserved words.
